lib/lte_link_control/lte_lc.c

```diff
@@ -900,7 +915,35 @@ int lte_lc_edrx_req(bool enable)
 		err = at_cmd_write(edrx_disable, NULL, 0, NULL);
 	}
 
-	return err;
+	if (err) {
+		LOG_ERR("Failed to %s eDRX, error: %d",
+			enable ? "enable" : "disable", err);
+		return err;
+	}
+
+	/* PTW must be requested after AT+CEDRXS is sent, and the length of the
+	 * string must be 4 to be valid.
+	 */
+	if (strlen(ptw_param) == 4) {
+		char ptw[25];
+		ssize_t len;
+
+		len = snprintf(ptw, sizeof(ptw),
+			       "AT%%XPTW=%d,\"%s\"", actt, ptw_param);
+		if ((len < 0) || (len > sizeof(ptw))) {
```

Comment:
It should actually check for `len >= sizeof(ptw)`, the return value indicates number of chars that would be written **not** counting the NULL terminator. So if it returns 25 in our case, it means it skipped the last character and put NULL in its place, resulting in malformed command.

Reply:
Good catch, should be fixed now